opal/util/output.c

```diff
@@ -782,9 +782,13 @@ static int open_file(int i)
             free(filename); /* release the filename in all cases */
             return OPAL_ERR_IN_ERRNO;
         }
-
+#if 0
```

Comment:
This prevents PMIx from deleting these files during cleanup (since they all end up in the PMIx directory in many situations).  We don't want to lose this functionality, but it makes the util / non-util OPAL split harder.  I think the right answer is yet another function pointer.  have an opal_pmix_opal_output_handler() function in pmix base that is set as the function pointer in the util/output.c code when the opal_init() call occurs (as opposed to the opal_init_util() cal).  We don't need this for utilities that aren't going to use PMIx, agree there.

Reply:
I'm afraid that is not enough, for any software that explicitly calls `opal_util_init` before `opal_init`. We could build a temporary list and once the RTE is initialized migrate all files there, but this seems extremely invasive.

Reply:
what if we didn't have libopen_pal.la include libopen_pal_util.la?  We can have libopen_pal.la include all the same source files as libopenpal_util.la, except for a single file which provided opal_pmix_register_cleanup() as an empty function?  A better solution would be weak symbols, but since Darwin doesn't support weak symbols, that's out.